mmdet/models/dense_heads/anchor_head.py

```diff
@@ -569,9 +572,18 @@ def get_bboxes(self,
             ]
             img_shape = img_metas[img_id]['img_shape']
             scale_factor = img_metas[img_id]['scale_factor']
-            proposals = self._get_bboxes_single(cls_score_list, bbox_pred_list,
-                                                mlvl_anchors, img_shape,
-                                                scale_factor, cfg, rescale)
+            if with_nms:
+                # some heads don't support with_nms argument
+                proposals = self._get_bboxes_single(cls_score_list,
+                                                    bbox_pred_list,
+                                                    mlvl_anchors, img_shape,
+                                                    scale_factor, cfg, rescale)
+            else:
+                proposals = self._get_bboxes_single(cls_score_list,
+                                                    bbox_pred_list,
+                                                    mlvl_anchors, img_shape,
+                                                    scale_factor, cfg, rescale,
+                                                    with_nms)
```

Comment:
If we just pass `with_nms` into `self._get_bboxes_single`, will any detector raise an error? 
I checked all heads inherited `AnchorHead`, it looks fine.

Reply:
`ATSSHead` will not raise an error, since it has own `get_bboxes`.
But `RPNHead` will raise an error.
The code above also avoids breaking compatibility with mmdetection forks and other PRs.
Needless to say, the code should be cleaned after the `with_nms` argument becomes standard.